src/System.Management.Automation/FormatAndOutput/common/ColumnWidthManager.cs

```diff
@@ -83,12 +83,19 @@ private bool AssignColumnWidths(Span<int> columnWidths)
             // we have columns with no width assigned
             // remember the columns we are trying to size
             // assign them the minimum column size
-            bool[] fixedColumn = new bool[columnWidths.Length];
+            int unfixedColumnCount = 0;
+            Span<bool> fixedColumn = stackalloc bool[columnWidths.Length];
             for (int k = 0; k < columnWidths.Length; k++)
             {
-                fixedColumn[k] = columnWidths[k] > 0;
                 if (columnWidths[k] == 0)
+                {
                     columnWidths[k] = _minimumColumnWidth;
+                    unfixedColumnCount++;
+                }
+                else
+                {
+                    fixedColumn[k] = true;
```

Comment:
It appears that stackalloc currently initializes mem to 0 but not sure how strong of a guarantee there is.  See https://stackoverflow.com/questions/8679052/initialization-of-memory-allocated-with-stackalloc  Is there enough of a perf benefit to skip assigning `false` to `fixedColumn[k]`?

Reply:
I went ahead and ran the [test you used in the issue](https://github.com/PowerShell/PowerShell/issues/15474#issuecomment-856119520) to see what the performance difference was.

Here are the results:

This is without assigning false:
```powershell
D:\Source\PowerShell [nick-15474 +1 ~1 -0 !]> $PSDefaultParameterValues['out-file:width'] = [int]::MaxValue - 1
D:\Source\PowerShell [nick-15474 +1 ~1 -0 !]> 1..500 | %{Measure-Command {Get-ChildItem Env:\ > env.log}} | Measure TotalMilliseconds -AllStats

Count             : 500
Average           : 30.7634942
Sum               : 15381.7471
Maximum           : 148.0802
Minimum           : 27.5581
StandardDeviation : 6.3614641496625
Property          : TotalMilliseconds
```

Assigning false:
```powershell
D:\Source\PowerShell [nick-15474 +1 ~1 -0 !]> $PSDefaultParameterValues['out-file:width'] = [int]::MaxValue - 1
D:\Source\PowerShell [nick-15474 +1 ~1 -0 !]> 1..500 | %{Measure-Command {Get-ChildItem Env:\ > env.log}} | Measure TotalMilliseconds -AllStats

Count             : 500
Average           : 30.3611324
Sum               : 15180.5662
Maximum           : 130.192
Minimum           : 26.7938
StandardDeviation : 5.19796767162906
Property          : TotalMilliseconds
```

It would appear that explicitly assigning false is more performant and avoids the issue you linked. I'll update the PR.

Reply:
zeroing is C# contract.

The high level test is not reliable. You should create C# test for a piece of the code and measure with BenchmarkDotNet.

Reply:
@iSazonov , I went ahead and created a [test ](https://gist.github.com/nickkimbrough/2c642ddd967ee5984a25506e6936586d) with BenchmarkDotNet (my first so hopefully good enough of a test). Between this and your SharpLab example, it would appear that the better solution is to skip assignment.

|            Method |     Mean |     Error |    StdDev |
|------------------ |---------:|----------:|----------:|
|    WithAssignment | 3.692 ns | 0.0067 ns | 0.0056 ns |
| WithoutAssignment | 3.458 ns | 0.0280 ns | 0.0248 ns |

Given this new data, I have reverted this commit.

Reply:
You could use `[Benchmark (Baseline =true)]` for one test to get a relative comparison in the report.

Reply:
I went ahead and did this and created another [gist](https://gist.github.com/nickkimbrough/bd4dbfb3aec3016519b4b4cf8539f12b). I also added in the current implementation we're looking at improving to get a measure:

|                    Method |       Mean |      Error |     StdDev |     Median | Ratio | RatioSD |
|-------------------------- |-----------:|-----------:|-----------:|-----------:|------:|--------:|
|            BeforeRefactor | 340.000 ns | 21.3756 ns | 59.5866 ns | 300.000 ns | 56.55 |    8.05 |
|    RefactorWithAssignment |   5.980 ns |  0.1370 ns |  0.1144 ns |   5.992 ns |  1.00 |    0.00 |
| RefactorWithoutAssignment |   5.021 ns |  0.0300 ns |  0.0266 ns |   5.023 ns |  0.84 |    0.02 |